src/net_processing.cpp

```diff
@@ -3563,32 +3563,44 @@ void ProcessMessage(
     return;
 }
 
+/** Maybe disconnect a peer and discourage future connections from its address.
+ *
+ * @param[in]   pnode           The node to check.
+ * @return                      True if the peer will be disconnected.
+ */
 bool PeerLogicValidation::MaybeDiscourageAndDisconnect(CNode& pnode)
 {
-    AssertLockHeld(cs_main);
-    CNodeState &state = *State(pnode.GetId());
+    {
+        LOCK(cs_main);
```

Comment:
```
net_processing.cpp:3574:9: error: acquiring mutex 'cs_main' that is already held [-Werror,-Wthread-safety-analysis]

        LOCK(cs_main);

        ^

Reply:
You might have to use clang++ to reproduce locally

Reply:
Ah. I didn't remove the lock annotation from the function declaration. Fixed.

Reply:
Saw this warning as well @ 5722b338d. Now building at 37e726579.